jina/clients/base/__init__.py

```diff
@@ -27,7 +27,12 @@ class BaseClient(ABC):
     :param kwargs: additional parameters that can be accepted by client parser
     """
 
-    def __init__(self, args: Optional['argparse.Namespace'] = None, **kwargs):
+    def __init__(
+        self,
+        args: Optional['argparse.Namespace'] = None,
+        results_as_docarray=False,
```

Comment:
@jacobowitz suggested I do this as the easiest solution, and I think it makes sense. Because to change the behaviour in Flow I changed PostMixin, which then also affects Client, but we want that to remain unchanged. So as I see it, either we split the Mixin somehow (which might defeats it purpose a little bit), or we make this optional on the Client level. That way the Flow can set this property for its internal Client. So that was the idea - is there a better way?

Reply:
Ok, then keep it okey